src/bufr2ncObsTypes.py

```diff
@@ -854,11 +854,16 @@ def extract_gpsro_obs_counts(self, bufr):
         # Mnemonic '(ROSEQ3)' returns the number of refractivity obs
         #
         NumBangleFreq = bufr.read_subset('{ROSEQ2}').astype(int).squeeze()
+        
+        # Steve: there are two checks we need to do here: 
+        # (all obs): reps of Q1 = reps of Q2 [NumROSEQ1 = NumBangleFreq] 
+        # (ref obs): NumRefrac = NumBangle
```

Comment:
I made these checks based on Jeff's gpsbufr2nc.py script, and after looking in the read_gps.f90 code I see they are different between the Python and Fortran code. The Fortran code seems overly cautious but perhaps the developers ran into a case where the libbufr.a code returned an array (the ubfint call with '{ROSEQ2}' mnemonic) with a different length than the number of ROSEQ1 replications (the ubfseq call with 'ROSEQ1' mnemonic). 

I think we should go with these changes since we likely won't be able to determine whether the situation mentioned above ('{ROSEQ2}' and 'ROSEQ1' are inconsistent) can happen. However, if you have reason to believe that the situation can't happen then I would vote to back these changes out.